### DISCUSSION TITLE
A note on a certain differential

### DISCUSSION THREAD

**Discussion by kellertuer** [JuliaManifolds/Manifolds.jl#461] [General]
As a continuation from a discussion on slack I started writing down an easy function @const-ae started with. Here is the derivation on why we have an original additional ``t`` in the derivative - the notebook is still missing the original `diff_check`. Feel free to comment on this state.
[🎈 note-constantin.jl — Pluto.jl.pdf](https://github.com/JuliaManifolds/Manifolds.jl/files/8146312/note-constantin.jl.Pluto.jl.pdf)

(somehow .jl files are not allowed as files here so here it is as a gist https://gist.github.com/kellertuer/3e17cfce6b0e6d40d3d8e7d14651323c )

**Comment by const-ae** [comment]
Hey, thanks a lot for getting started on this. I have forked the notebook (https://gist.github.com/const-ae/ae5e26b255978a3b4eaec19f493282f8) and added the code for the `check_diff` function. 
The rendered pdf of the notebook is at [2022-02-28_note_diff_v2.pdf](https://github.com/JuliaManifolds/Manifolds.jl/files/8152537/2022-02-28_note_diff_v2.pdf).


I have also added code for the Rotation manifold and found that for random points on the tangent space, the differential that you suggest does not converge quadratically, whereas the version with `βdifferential_exp_arg_with_time` does. I will investigate this further to better understand what could explain the difference or if this might still just be an artifact.

**Reply by kellertuer** [reply]
Interesting, I will take.a closer look - the t-scaled one looks to maybe be numerically faster but theoretically that still seems strange.

**Comment by kellertuer** [comment]
I found it!

It was a little tricky and I confused myself with the (little older) notation in Manopt compared to Manifolds.jl and we should at some point rename two differentials (next breaking change).

TLDR: We are both right. The thing is that you tried to use a differential for something it was not meant to be used for. And it is a chain rule.

Long story: (Trying without math but with code, since math is not rendered here)

There are two ways to specify a geodesic: The first is done with a start point `p` and an end point `q`; these are called `shortest_geodesic(M,p,q)`s in Manifolds.jl In Manopt.jl the differential is still called `differential_geodesic_startpoint` (Dp) and `_endpoint`(Dq) of the shortest geodesics

The second way to specify a geodesic is by a start point `p` and a direction `X`(called `geodesic(M,p,X)` in Manifolds.jl). The formula here for Sid curve just reads `c(t) = exp(M, p t*X)`. Here Dp of the geodesic would just be Dp exp (exponential with respect to base point) at t*X with direction Y.

**But** the differential of the geodesic with respect to the argument introduces a chain rule that leads to exactly the terms you specified (even “with time” is not that bad for a name).
While on the other hand DX exp does not have the t dependency, since t is constant 1. This is also why all code and formulae from J. Persch for example are correct.

One could – still – use/extend the given coefficients with the t, since for DX of exp(M,p,X) we anyways plug in t=1 and then your new coefficient agree with the old ones; I adapted the documentation and code accordingly, which does not break any of the calls (was called with t=1 only until now anyways)

I already started the result as a PR in https://github.com/JuliaManifolds/Manopt.jl/pull/114, where I also denoted the few small caveats still to be taken into account.

I hope the explanations are understandable.

**Reply by const-ae** [reply]
Great that you found the source of the disagreement and were able to resolve this :) I am very happy to see that my tinkering around helped to improve `Manopt.jl`.

**Reply by kellertuer** [reply]
Well I still have to find time to finish this, but indeed it makes Manopt.jl more consistent with Manifolds.jl – so thanks for that.
